apps/meteor/app/livechat/server/lib/Livechat.js

```diff
@@ -1367,7 +1367,9 @@ export const Livechat = {
 
 		const room = LivechatRooms.findOneById(rid);
 
-		const visitor = LivechatVisitors.getVisitorByToken(token, { fields: { _id: 1, token: 1, language: 1, username: 1, name: 1 } });
+		const visitor = Promise.await(
+			LivechatVisitors.getVisitorByToken(token, { fields: { _id: 1, token: 1, language: 1, username: 1, name: 1 } }),
+		);
```

Comment:
Is it the case that visitor was not returning quickly enough, and therefore null, and that somehwere else it was defaulting to 'agent'?  Maybe in Messages.createTranscriptHistoryWithRoomIdMessageAndUser?